File: special_pythagorean_triplet/java/special_pythagorean_triplet.java
public class special_pythagorean_triplet {
	public static void main(String[] args) {
		int goalSum=1000;
		for (int a=1; a<goalSum/3; a++) {
			for (int b=a+1; b<goalSum/2; b++) {
				int c=goalSum-a-b;
				if (c>0 && ((a*a)+(b*b)) == (c*c)) {
					if (a+b+c==goalSum) {
						System.out.println("a="+a+" b="+b+" c="+c+" product="+(a*b*c));
					}
				}
			}
		}
	}
}

File: special_pythagorean_triplet/php/special_pythagorean_triplet.php
<?php

$goalSum=1000;

$start = microtime(true);

for ($a=1; $a < $goalSum/3; $a++) { 
	for ($b=$a+1; $b < $goalSum/2; $b++) { 
		$c = $goalSum-$a-$b;
		if ($c > 0 && (($a*$a) + ($b*$b)) == ($c*$c)) {
			if ($a+$b+$c == $goalSum) {
				$product=$a*$b*$c;
				echo "a={$a} b={$b} c={$c} product={$product}\n";
				echo "Took ".(microtime(true)-$start)." seconds";
				die();
			}
		}
	}
}

